src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -1995,6 +1995,240 @@ public async Task GetAsync_ExpectContinueTrue_NoContent_StillSendsHeader()
             });
         }
 
+        public static IEnumerable<object[]> Interim1xxStatusCode()
+        {
+            yield return new object[] { (HttpStatusCode) 100 }; // 100 Continue.
+            // 101 SwitchingProtocols will be treated as a final status code.
+            yield return new object[] { (HttpStatusCode) 102 }; // 102 Processing.
+            yield return new object[] { (HttpStatusCode) 103 }; // 103 EarlyHints.
+            yield return new object[] { (HttpStatusCode) 150 };
+            yield return new object[] { (HttpStatusCode) 180 };
+            yield return new object[] { (HttpStatusCode) 199 };
+        }
+
+        [Theory]
+        [MemberData(nameof(Interim1xxStatusCode))]
+        public async Task SendAsync_1xxResponsesWithHeaders_InterimResponsesHeadersIgnored(HttpStatusCode responseStatusCode)
+        {
+            // Skip test on .NET Framework since it doesn't have the fix.
+            if (PlatformDetection.IsFullFramework && (int)responseStatusCode >= 102) return;
+
+            var clientFinished = new TaskCompletionSource<bool>();
+            const string TestString = "test";
+            const string CookieHeaderExpected = "yummy_cookie=choco";
+            const string SetCookieExpected = "theme=light";
+            const string ContentTypeHeaderExpected = "text/html";
+
+            const string CookieHeaderIgnored1 = "ignore_cookie=choco1";
+            const string CookieHeaderIgnored2 = "ignore_cookie=choco2";
+            const string SetCookieIgnored1 = "hello=world";
+            const string SetCookieIgnored2 = "net=core";
+            const string ContentTypeHeaderIgnored1 = "text/xml";
+            const string ContentTypeHeaderIgnored2 = "text/plain";
```

Comment:
Do these all need to be defined as consts?  Can't most of them, like this one, just be text used where the const is referenced?

Reply:
This one (`ContentTypeHeaderIgnored`) and `CookieHeaderIgnored` can just be text, but I think giving the value a variable name can serve as comment/document to explain the purpose, and make the string interpolation below clear.

My preference is to remain using the consts, because for test project I think detail is preferred over optimization.

Reply:
> My preference is to remain using the consts, because for test project I think detail is preferred over optimization.

I agree it's not worth optimizing, but in this case I wasn't suggesting it due to optimization... it's only used in one place, so just using it inline in that one place makes it easier to read IMHO.

Reply:
Fixed in new commit.